### ISSUE TITLE
Meta - ComponentConfig scaffolds should not be done by default 

### ISSUE THREAD

**Issue by camilamacedo86** [kubernetes-sigs/kubebuilder#2782]
### What broke? What's expected?

Users should have the scaffold of https://github.com/kubernetes-sigs/kubebuilder/blob/master/docs/book/src/component-config-tutorial/testdata/project/config/manager/controller_manager_config.yaml ONLY/when they create the project following the tutorial: https://book.kubebuilder.io/component-config-tutorial/api-changes.html

That means, the scaffold should only be done when we run `kubebuilder init --domain tutorial.kubebuilder.io --component-config`. However, it has been added to the default scaffolds when we run `kubebuilder init` and it has brought confusion since users do not understand how properly use it and that their code in the main.go also needs to be customized accordingly. 

### Reproducing this issue

**Examples:**
- https://github.com/kubernetes-sigs/kubebuilder/pull/2778
- https://github.com/operator-framework/operator-sdk/issues/5884

### KubeBuilder (CLI) Version

3.5.0

### Plugin versions

Therefore, these options would fit in a plugin and should not be a flag. It is only a flag because it was implemented in the past (Kubebuilder 2.X) before the plugin design was introduced. 

- [ ] **Proposed changes (Bug FIX)**
a) Ensure that the component config scaffolds are not done by default and are only done when the flag is used
b) Review the documentation and ensure that it proper clarifies how it should be used and work

- [ ] **Proposed changes in the long term (RFE)**
a) Deprecated the flag
b) Move ComponentConfig behaviour to a plugin that can be used in the init/edit subcommands.

**Comment by camilamacedo86** [comment]
Re-open this one it still missing we create a plugin for this feature and deprecate the flag. 
Also, we should update the docs in the pr where it will be addressed.

**Comment by laxmikantbpandhare** [comment]
@camilamacedo86 yes, working on plug-in part for this command.

**Comment by laxmikantbpandhare** [comment]
@camilamacedo86 - Raised PR #2942 for one of the fixes.

Working on the below part and will raise the PR soon.

`Move ComponentConfig behavior to a plugin that can be used in the init/edit subcommands.`

**Comment by laxmikantbpandhare** [comment]
@camilamacedo86 raised PR for `init/edit` subcommands: https://github.com/kubernetes-sigs/kubebuilder/pull/3022

**Comment by rustycl0ck** [comment]
- Just mentioning here for information: https://github.com/kubernetes-sigs/controller-runtime/issues/895
  There might be some useful information in this discussion about how to move forward with this.

I stumbled upon this issue because the linter is now throwing errors that [`options.AndFrom`](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.15.0/pkg/manager#Options.AndFrom) is deprecated and will be removed in a future version. Will have to bypass the linter for now, but the documents should at least be updated soon to discourage people from migrating to `ComponentConfig` now (maybe can point to the above thread for more information about alternatives or discussions in the meantime).

**Comment by camilamacedo86** [comment]
Yes, we need to just deprecated the flag and we deprecated the configuration already, see: https://github.com/kubernetes-sigs/kubebuilder/pull/3442

@laxmikantbpandhare could you please rebase your PR https://github.com/kubernetes-sigs/kubebuilder/pull/3023, ensure that we are deprecating it for go/v4 and help us to get it merged ?

**Comment by laxmikantbpandhare** [comment]
@camilamacedo86 - I was on leave and away from coding. I will look into it and merge it ASAP.

**Comment by aperullo** [comment]
What is the intended replacement for componentconfig we should be using instead?
